src/discounts/filters/useUrlValueProvider.ts

```diff
@@ -0,0 +1,110 @@
+import { InitialStateResponse } from "@dashboard/components/ConditionalFilter/API/InitialStateResponse";
+import {
+  FilterContainer,
+  FilterElement,
+} from "@dashboard/components/ConditionalFilter/FilterElement";
+import { FilterValueProvider } from "@dashboard/components/ConditionalFilter/FilterValueProvider";
+import { TokenArray } from "@dashboard/components/ConditionalFilter/ValueProvider/TokenArray";
+import { UrlEntry } from "@dashboard/components/ConditionalFilter/ValueProvider/UrlToken";
+import { stringify } from "qs";
+import { useEffect, useState } from "react";
+import useRouter from "use-react-router";
+
+type Structure = Array<string | UrlEntry | Structure>;
+
+const prepareStructure = (filterValue: FilterContainer): Structure =>
+  filterValue.map(f => {
+    if (typeof f === "string") {
+      return f;
+    }
+
+    if (Array.isArray(f)) {
+      return prepareStructure(f);
+    }
+
+    return f.asUrlEntry();
+  });
+
+export const useUrlValueProvider = (
+  locationSearch: string,
+): FilterValueProvider => {
+  const router = useRouter();
+  const params = new URLSearchParams(locationSearch);
+  const [value, setValue] = useState<FilterContainer>([]);
+
+  const activeTab = params.get("activeTab");
+  const query = params.get("query");
+  const before = params.get("before");
+  const after = params.get("after");
+  params.delete("asc");
+  params.delete("sort");
+  params.delete("activeTab");
+  params.delete("query");
+  params.delete("before");
+  params.delete("after");
+
+  const tokenizedUrl = new TokenArray(params.toString());
+
+  useEffect(() => {
+    const emptyInitialState = InitialStateResponse.empty();
+    emptyInitialState.attribute = {
+      startDate: {
+        choices: [],
+        inputType: "DATE_TIME",
+        label: "Start date",
+        slug: "start-date",
+        value: "start-date",
+      },
+      endDate: {
+        choices: [],
+        inputType: "DATE_TIME",
+        label: "End date",
+        slug: "end-date",
+        value: "end-date",
+      },
+    };
+
+    setValue(tokenizedUrl.asFilterValuesFromResponse(emptyInitialState));
+  }, [locationSearch]);
```

Comment:
This causes instant re-render once once you render that component. additionally you directly mutate the fields. Additionally, if the empty state is different for discounts you can create dedicated named constructor for it ` InitialStateResponse.discountsEmpty()` or why just don't add these fields to be supported?